live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/messaging/in/mapper/RepoEventMapper.java

```diff
@@ -67,11 +69,63 @@ public IngestNewNodeEvent mapToIngestNewNodeEvent(RepoEvent<DataAttributes<NodeR
                         node.isFile(),
                         node.isFolder(),
                         node.getCreatedAt().toInstant().toEpochMilli(),
-                        node.getProperties()
-                                .entrySet()
-                                .stream()
-                                .filter(property -> Objects.nonNull(property.getKey()) && Objects.nonNull(property.getValue()))
-                                .map(property -> new NodeProperty<>(property.getKey(), property.getValue()))
-                                .collect(Collectors.toSet())));
+                        getNodeCustomProperties(node)));
+    }
+
+    public UpdateNodeEvent mapToUpdateNodeMetadataEvent(RepoEvent<DataAttributes<NodeResource>> event)
+    {
+        return new UpdateNodeEvent(
+                event.getTime().toInstant().toEpochMilli(),
+                calculateFieldDelta(event, NodeResource::getName),
+                calculateFieldDelta(event, NodeResource::getPrimaryAssocQName),
+                calculateFieldDelta(event, NodeResource::getNodeType),
+                calculateFieldDelta(event, this::getModifiedByUserWithId),
+                calculateFieldDelta(event, NodeResource::getAspectNames),
+                calculateFieldDelta(event, NodeResource::isFile),
+                calculateFieldDelta(event, NodeResource::isFolder),
+                calculateFieldDelta(event, this::getNodeCustomProperties));
```

Comment:
This is all really neat - I like it. I think that you might need to treat the custom properties differently as I think the delta works differently for this one field. For example
```
    "resourceBefore": {
      "@type": "NodeResource",
      "modifiedAt": "2021-01-21T11:14:25.223Z",
      "properties": {
        "cm:title": null,
        "cm:taggable": null,
        "cm:description": null
      },
      "aspectNames": [
        "cm:versionable",
        "cm:author",
        "cm:thumbnailModification",
        "cm:titled",
        "rn:renditioned",
        "cm:auditable"
      ]
    },
```
means that the "modifiedAt" date changed, the list of aspect names changed and three of the properties changed. The `aspectNames` list is the full old list, but the `properties` map is only some of the old custom properties.